### PR TITLE
13291 hide channels

### PR DESCRIPTION
**Pull request by endulab**
### What does the PR do

Extend chats model with channel permissions info and hideIfPermissionsNotMet.
Visibility of chat item is based on: member roles, channel permissions, hideIfPermissionsNotMet.
If all channels from category are hidden, category item is also hidden.
If all channels in community are hidden, information label is displayed.

Related status-go [pr](https://github.com/status-im/status-go/pull/4884)

Issue #13291 

https://github.com/status-im/status-desktop/assets/61889657/6cc6ada3-f811-45a9-b252-13bf91171033

### CODE REVIEW

File: src/app/modules/main/chat_section/model.nim [hunk 1]
```diff
@@ -227,6 +263,18 @@ QtObject:
         defer: index.delete
         self.dataChanged(index, index, @[ModelRole.CategoryOpened.int])
 
+  proc changeCategoryItemIfHiddenBecauseNoPermitted*(self: Model, id: string) =
```
Comment (jrainville) [inline src/app/modules/main/chat_section/model.nim:266]: I guess you made a typo?
```suggestion
  proc changeCategoryItemIsHiddenBecauseNotPermitted*(self: Model, id: string) =
```

File: src/app/modules/main/chat_section/model.nim [hunk 2]
```diff
@@ -227,6 +263,18 @@ QtObject:
...
+    if id == "":
+      return
+    let index = self.getItemIdxById(id)
+    if index == -1:
+      return
+    if not self.items[index].isCategory:
+      return
+    let modelIndex = self.createIndex(index, 0, nil)
+    defer: modelIndex.delete
+    self.dataChanged(modelIndex, modelIndex, @[ModelRole.ShouldBeHiddenBecausePermissionsAreNotMet.int])
```
Comment (jrainville) [inline src/app/modules/main/chat_section/model.nim:276]: You didn't change anything actually. I think you forgot to change the value.

Also, we only want to hide the category if all the channels in it are hidden. So we need a loop to check the channels in the category here.

**Review by jrainville — COMMENTED** [review]
Good job

File: ui/app/AppLayouts/Chat/views/ChatView.qml [hunk 3]
```diff
@@ -234,6 +240,18 @@ StatusSectionLayout {
...
     }
 
+    Component {
+        id: allChatsAreHiddenComponent
+
+        StatusBaseText {
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            visible: root.allChannelsAreHiddenBecauseNotPermitted
+            text: qsTr("Sorry, you don't hodl the necessary tokens to view or post in any of <b>"+sectionItemModel.name+"</b> channels")
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Chat/views/ChatView.qml:250]: Even:
```qml
text: qsTr("Sorry, you don't hodl the necessary tokens to view or post in any of <b>%1</b> channels").arg(sectionItemModel.name)
```

[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Chat/views/ChatView.qml:250]: Still needs fixing @endulab

File: src/app/modules/main/chat_section/model.nim (hunk 2 above)
Comment (endulab) [reply src/app/modules/main/chat_section/model.nim:289]: I renamed this function and also added a few comments for better understanding.
Every item has ShouldBeHiddenBecausePermissionsAreNotMet role. It is a special, helper role that I added which is a combination of other roles. This role is used to filter in QML. The value of this role is computed by itemShouldBeHiddenBecauseNotPermitted() function for both normal items and section items.
Whenever data with roles: HideIfPermissionsNotMet, ViewOnlyPermissionsSatisfied, ViewAndPostPermissionsSatisfied, are changed, also we trigger refreshing data with ShouldBeHiddenBecausePermissionsAreNotMet role.
This works correctly as I showed on the demo.

**Pull request merged**